components/director/internal/model/formation_assignment.go

```diff
@@ -37,6 +37,22 @@ type FormationAssignmentPage struct {
 	TotalCount int
 }
 
+// FormationAssignmentState represents the possible states a formation assignment can be in
+type FormationAssignmentState string
+
+const (
+	// InitialAssignmentState indicates that nothing has been done with the formation assignment
+	InitialAssignmentState FormationAssignmentState = "INITIAL"
```

Comment:
Ir will be easier for maintenance if the constant names use the following naming schema:
<Object><State>
In this case names to be:
AssignmentStateInitial
AssignmentStateReady
AssignmentStateConfigPending
...
As when you want to reference the object state in code will use autocomplete to look for the object and will see the states that are applicable for this object.